File: bench/langs/Ledger.java
package bench;

/** Keeps the running balance for one account. */
public class Ledger {
    static final int LIMIT = 5000;

    private final String owner;

    public Ledger(String owner) {
        this.owner = owner;
    }

    /** Bills the account and returns what is left. */
    public int charge(int amount) {
        if (amount > LIMIT) {
            return refuse(amount);
        }
        return commit(amount);
    }

    private int refuse(int amount) {
        warn(owner);
        return 0;
    }

    private int commit(int amount) {
        write(owner, amount);
        return amount;
    }
}

File: bench/langs/Ledger.kt
package bench

/** Keeps the running balance for one account. */
class Ledger(private val owner: String) {

    /** Bills the account and returns what is left. */
    fun charge(amount: Int): Int {
        if (amount > LIMIT) {
            return refuse(amount)
        }
        return commit(amount)
    }

    private fun refuse(amount: Int): Int {
        warn(owner)
        return 0
    }

    private fun commit(amount: Int): Int {
        write(owner, amount)
        return amount
    }

    companion object {
        const val LIMIT = 5000
    }
}
